### ISSUE TITLE
random seed

### ISSUE THREAD

**Issue by Lili-Cao** [gnina/gnina#256]
### Issue summary
Hi, 

I am wondering why the gnina results are different when I use the same parameters and run many replicas.
I didn't set a value to the random seed. (I assued it there is a default one? some place says it's 42)

Therefore I am wondering, if I don't set the random seed parameter, does gnina always using a default seed?
If so, why is the result not reproducible and the performance differ a lot?

Thanks!
Lili

### Your system configuration
Operating system: Linux
Compiler:
CUDA version (if applicable):
CUDNN version (if applicable):
Python version:

**Comment by dkoes** [comment]
No, it does not use a default value for the seed.  In fact, you should see it print out the value of the random seed it is using and it will be different for each run unless you explicitly set it.

**Comment by Lili-Cao** [comment]
thanks for your prompt reply!
Aha, that's the reason.

Due to the way I integrated Gnina and the way of logging results,  I can't see the output info generated while Gnina running. 
Could you please tell me which numbers Gnina would use for a random seed?
So I can run Gnina with all the seeds and try to reproduce my results?

Thanks!

**Comment by dkoes** [comment]
If you don't set the seed then it is the id of the process times the current time (modulo 32 bits).  So you really have no hope of guessing what seed was used in a run unless you saved the output or explicitly set the seed.

**Comment by Lili-Cao** [comment]
: (  
Em...  OK, seems I have to set it explicitly then. Thank you!

**Comment by Lili-Cao** [comment]
Let's say if I run gnina with exhaustiveness = 8 and poses = 8, but with 4 different seeds, and take the best prediction.
and run exhaustiveness =64 and poses = 8, only one seed.
Which one do you think would possibly give better prediction?

**Comment by dkoes** [comment]
4*8 = 32 < 64 so the later since you are doing twice as much sampling.  If you do the same amount of sampling the results should be the same in expectation.

**Comment by Lili-Cao** [comment]
The reason I am interested in this is that, I have tried different options for exhaustiveness and poses.
And I found exhaustiveness = 8 and poses = 16 have better predictions than exhaustiveness =64 and poses = 8  exhaustiveness =64 and poses = 16

**Comment by dkoes** [comment]
They won't be exactly the same because the random number sequence will be different.
